### DISCUSSION TITLE
Type definition for the configuration object

### DISCUSSION THREAD

**Discussion by smellai** [uNmAnNeR/imaskjs#972] [Q&A] [answered]
**What problem you are trying to solve?**
I'm defining an object with the config parameters for `useIMask`, in order to be able to share it. Is there any exposed type definition that I can use and attach to it? Sorry, I can find any doc about that, neither find out what to use while exploring react-imask source code.

**Describe the solution you'd like**
```
const getIMaskConfig = (format: hourFormat) => ({
  mask: format === '24' ? 'HH:mm' : 'HH:mm AA',
  placeholderChar: '-',
  lazy: false,
  overwrite: true,
  prepare: (value: string) => value.toLowerCase(),
  blocks: {
    HH: {
      mask: MaskedRange,
      from: 0,
      to: format === '24' ? 23 : 12,
      maxLength: 2,
      autofix: 'pad',
    },
    mm: {
      mask: MaskedRange,
      from: 0,
      to: 59,
      maxLength: 2,
      autofix: 'pad',
    },
    AA: {
      mask: MaskedEnum,
      enum: ['AM', 'am', 'PM', 'pm', 'aM', 'Am', 'pM', 'Pm'],
      maxLength: 2,
    },
  },
} satisfies ConfigType);
```

I'm wondering if I can import anything to be used as **ConfigType**

**Comment by uNmAnNeR** [comment, marked as answer]
seems like you need
https://imask.js.org/api/types/react_imask.ReactMaskOpts.html

**Comment by smellai** [comment]
Thanks, but it looks like is not exported?

**Reply by uNmAnNeR** [reply]
@smellai  it is exported https://github.com/uNmAnNeR/imaskjs/blob/da541c6f8f621ded0e03ff3ab9225b638d1c4438/packages/react-imask/src/index.ts#L13

**Comment by smellai** [comment]
sorry I was in older version, thanks!
